api-examples/pytorch_cpp/classify-text.cpp

```diff
@@ -0,0 +1,133 @@
+#include <torch/script.h>
+#include "json.hpp"
+using json = nlohmann::json;
+
+#include <iostream>
+#include <fstream>
+#include <memory>
+#include <string>
+#include <algorithm>
+#include <glob.h>
+
+#define UNK 3
+#define PAD 0
+
+
+void lower_case(const std::string& word, std::string& dst) {
+    dst.resize(word.size());
+    std::transform(
+        word.begin(),
+        word.end(),
+        dst.begin(),
+        ::tolower
+    );
+}
+
+
+void find_vocab_file(const std::string& bundle, const std::string& vocab, std::string& dst) {
+    glob_t glob_result;
+    std::stringstream file_name;
+    file_name << bundle << "/vocabs-" << vocab << "-*.json";
+    glob(file_name.str().c_str(), GLOB_TILDE, NULL, &glob_result);
+    dst = glob_result.gl_pathv[0];
+}
+
+
+int main(int argc, char** argv) {
+    if (argc < 3) {
+        std::cerr << "usage: classify-text export_bundle tokens...\n";
+        return -1;
+    }
+
+    std::string bundle(argv[1]);
+    std::vector<std::string> args(argv + 2, argv + argc);
+
+    std::stringstream model_file;
+    model_file << bundle << "/model.pt";
+    std::shared_ptr<torch::jit::script::Module> model = torch::jit::load(model_file.str());
```

Comment:
wouldnt `std::unique_ptr` work here?  Im not seeing an obvious need to have a smart ptr

Reply:
`std::shared_ptr` is what it was used in example code in the pytorch tutorial. I tried using `std::unique_ptr` and it wouldn't compile.